### PR TITLE
feat: Add progress callback

### PR DESCRIPTION
**Pull request by pajowu**
## What does this PR address?

This PR adds a binding for the progress callback added to whispercpp in https://github.com/ggerganov/whisper.cpp/pull/600

## Before submitting:

- [x] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [x] Did you read through and follow [development guidelines](../DEVELOPMENT.md)?
- [x] Did your changes require updates to the documentation? Have you updated those accordingly?
- [x] Did you write tests to cover your changes?

### CODE REVIEW

File: tests/export_test.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
     import numpy as np
     import ffmpeg
     from numpy.typing import NDArray
+    from typing import Any
```
Comment (aarnphm) [inline tests/export_test.py:15]:
```suggestion
```

**Comment by aarnphm** [comment]
Hmm something is failing

**Comment by pajowu** [comment]
I'll have another look what went wrong

**Comment by pajowu** [comment]
I forgot to bump the whispercpp dependency. Now everything works

File: rules/deps.bzl [hunk 2]
```diff
@@ -74,7 +74,7 @@ def internal_deps():
         init_submodules = True,
         recursive_init_submodules = True,
         remote = "https://github.com/ggerganov/whisper.cpp.git",
-        commit = "09e90680072d8ecdf02eaf21c393218385d2c616",
+        commit = "2f889132c66051b14c6f8770e9b3d4e3f159821d",
```
Comment (aarnphm) [inline rules/deps.bzl:77]: cc @aarnphm: Time to write a script to bump this along side with submodules update

**Review by aarnphm — APPROVED** [review]
LGTM. I will probably release a new version this weekend.

**Comment by sorgfresser** [comment]
Awesome feature, thanks a lot for adding! Is there a way we can get a wheel for this or do I have to use `pip install git+https://github.com/aarnphm/whispercpp.git` ?

**Comment by pajowu** [comment]
There isn't yet, as no release was published yet. I also just use the git version without wheels

**Comment by aarnphm** [comment]
I'm current on leave atm and will be back around april 20th

**Pull request merged**